src/vm_pages.rs

```diff
@@ -1131,38 +1132,57 @@ impl<'a, T: GuestStagePagingMode> FinalizedVmPages<'a, T> {
         if !state_pages.is_contiguous() {
             return Err(Error::NonContiguousPages);
         }
-        let vcpu_pages = self.get_converted_pages(vcpus_addr, num_vcpu_pages)?;
-        if !vcpu_pages.is_contiguous() {
-            return Err(Error::NonContiguousPages);
-        }
         let id = self
             .inner
             .page_tracker
             .add_active_guest()
             .map_err(Error::GuestId)?;
-
         let guest_root_pages =
             SequentialPages::from_pages(self.assign_state_pages_for(guest_root_pages, id)).unwrap();
         let guest_root =
             GuestStagePageTable::new(guest_root_pages, id, self.inner.page_tracker.clone())
                 .unwrap();
 
-        let mut state_pages = self.assign_state_pages_for(state_pages, id);
-        let box_page = state_pages.next().unwrap();
-        let region_vec_pages = SequentialPages::from_pages(state_pages).unwrap();
+        let mut state_pages =
+            SequentialPages::from_pages(self.assign_state_pages_for(state_pages, id)).unwrap();
+        let vm_state_pages = state_pages
+            .take(TVM_VM_STATE_PAGES)
+            .ok_or(Error::NotEnoughMemory)?;
```

Comment:
This leaks `state_pages` on error. We should check that we have enough pages up front.

Reply:
Same as above.